lib/ex_aws/s3.ex

```diff
@@ -488,22 +489,54 @@ defmodule ExAws.S3 do
 
   Handles initialization, uploading parts concurrently, and multipart upload completion.
 
-  ## Examples
+  ## Uploading a stream
+
+  Streams that emit binaries may be uploaded directly to S3. Each binary will be uploaded
+  as a chunk, so it must be at least 5 megabytes in size. The `S3.Upload.stream_file`
+  helper takes care of reading the file in 5 megabyte chunks.
   ```
   "path/to/big/file"
-  |> S3.Upload.stream_file!
+  |> S3.Upload.stream_file
   |> S3.upload("my-bucket", "path/on/s3")
-  |> ExAws.request! #=> {:ok, :done}
+  |> ExAws.request! #=> :done
   ```
 
-  ## Notes
+  ## Uploading a flow
+
+  GenStage Flows can also be uploaded directly to S3. However, due to the concurrent
+  nature of flows, the producer must emit events in the following format
+  `{binary, index}` to ensure they are uploaded in the correct order. Binaries
```

Comment:
> However, due to the concurrent nature of flows, the producer must emit events in the following format `{binary, index}` to ensure they are uploaded in the correct order.

The chunks can be uploaded in arbitrary order. Rather, the indices are used by AWS to determine in what order to combine the pieces when completing the result. I would say:

" All parts uploaded to AWS must be tagged with a `1` based index so that they can be assembled in order when finalizing the multi part upload."